File: android/app/src/main/java/com/yano/interplan/imble3_app/ContextHolder.java
package com.yano.interplan.imble3_app;


import android.app.Activity;
import android.content.Context;

import com.facebook.react.bridge.ReactApplicationContext;

/**
 * React Nativeコンテキストの受け渡し
 */
public class ContextHolder {
    private static ReactApplicationContext applicationContext;

    public static void init(ReactApplicationContext context) {
        applicationContext = context;
    }
    public static Activity getActivity(){
        if (applicationContext == null) {
            throw new IllegalStateException("activity is null");
        }
        return applicationContext.getCurrentActivity();
    }

    public static ReactApplicationContext getContext() {
        if (applicationContext == null) {
            throw new IllegalStateException("ContextHolder is not initialized. Call init() with application context first.");
        }
        return applicationContext;
    }
}

File: android/app/src/main/java/com/yano/interplan/imble3_app/ContextHolder.kt
package com.yano.interplan.imble3_app

import android.app.Activity
import com.facebook.react.bridge.ReactApplicationContext


/**
 * React Nativeコンテキストの受け渡し
 */
object ContextHolder {
    private var applicationContext: ReactApplicationContext? = null

    fun init(context: ReactApplicationContext?) {
        applicationContext = context
    }

    val activity: Activity?
        get() {
            checkNotNull(applicationContext) { "activity is null" }
            return applicationContext!!.currentActivity
        }

    val context: ReactApplicationContext
        get() {
            checkNotNull(applicationContext) { "ContextHolder is not initialized. Call init() with application context first." }
            return applicationContext!!
        }
}
